lib/WUI/nhttp/gcode_upload.cpp

```diff
@@ -292,7 +310,9 @@ UploadHooks::Result GcodeUpload::finish(const char *final_filename, bool start_p
     // Close the file first, otherwise it can't be moved
     ftruncate(fileno(tmp_upload_file.get()), ftell(tmp_upload_file.get()));
     tmp_upload_file.reset();
-    return try_rename(fname, final_filename, [&](char *filename) -> UploadHooks::Result {
+    auto putParams = std::get_if<PutParams>(&upload);
+    bool overwrite = putParams and putParams->overwrite;
```

Comment:
Style nitpick:

* Using of `and` and `or` in C++ is some new kind of decadence (or, actually, they seem to be ancient and never have reached any kind of widespread use :-O … anyway, they seem unusual). When trying to find out why, it seems they tend to be a bit confusing, because `and` is equivalent to `&&`, while `and_eq` is equivalent to `&=` which acts differently.
* We prefer to _explicitly_ compare pointers to `nullptr` in contrast to treating them as bools implicitly.